### DISCUSSION TITLE
[Feature]: Cloud Storage File Preview

### DISCUSSION THREAD

**Discussion by yougotwill** [mikekazakov/nimble-commander#340] [Ideas]
### Confirmation

- [X] I have read the contributing guidelines

### Version of Nimble Commander

Version 1.6.0 (4087), macOS 14.5 (23F79)

### Distribution Type

Downloaded directly from the website (including Brew)

### Expected Result

*I use Google Drive but I think this applies to any cloud storage provider*

If I have a file stored in the cloud that is not saved to my device you cannot preview it. The preview is empty.

Opening the file will automatically download it and open it. I would like the same behaviour if we trigger the preview please 🙏

### Actual Result

Preview fails and just shows the file metadata.

### Steps to reproduce

1. Have a file on your cloud provider that you have not downloaded to your device. 
2. Navigate to cloud folder in Nimble Commander
3. Use `Show Preview` command

### Additional Information

# Video

https://github.com/user-attachments/assets/c034fac0-61e4-49f3-a721-fe2d41d105b9

**Comment by yougotwill** [comment]
Alternatively copy what finder does.

1. Show a cloud icon next to the file that is not saved to the device.
2. Have a command (that we can assign a hotkey to) to download the file to our device which then refreshes the preview.

**Comment by mikekazakov** [comment]
Hi, thanks for bringing attention to this.
Unfortunately, I'm not aware how to implement it and likely won't have bandwidth to work on this feature.
If anybody wants to work on it - pull requests are welcome.
